platooning_strategic/src/platoon_strategic.cpp

```diff
@@ -169,6 +169,9 @@ namespace platoon_strategic
             single_lane_road_ = false;
             ROS_DEBUG_STREAM("Vehicle is NOT in a single-lane road");
         }
+
+        // Store the current local lane index of the host vehicle (0 is rightmost lane, 1 is second rightmost, etc.); considers only the current travel direction
+        current_lane_index_ = (routing_graph->rights(current_lanelet[0])).size();
```

Comment:
Seems a bit tacked on here, especially considering that it's a bit strange for a function named `check...` or `get...` to be updating object state under the hood. I could imagine that when making edits to this class later someone might not ensure that `checkForRightMostLane()` is called before the message publication to ensure that the value of this variable is set correctly. 

It's here though because all the needed data elements are already present in this function right? Would it be a big lift to extract this to a separate function `getCurrentLaneIndex()` or something?

Reply:
Agreed, I've gone ahead and extracted this into its own function `getCurrentLaneIndex()`.